lib/cdo/secrets.rb

```diff
@@ -65,8 +65,8 @@ def get(key)
       @values[key] ||= client_promise.then do |client|
         parse_json(get_secret_value(client, key))
       rescue => exception
-        exception.message << " Key: #{key}"
-        raise
+        modified_message = "#{exception.class}: #{exception.message} Key: #{key}"
+        raise exception.class.new(modified_message)
```

Comment:
this won't always work:
```
[development] dashboard > Aws::S3::Errors::NotFound.new('message')
/Users/dsb/.rbenv/versions/3.0.5/lib/ruby/gems/3.0.0/gems/aws-sdk-core-3.180.0/lib/aws-sdk-core/errors.rb:17:in `initialize': wrong number of arguments (given 1, expected 2..3) (ArgumentError)
```
(apologies if you received this twice)

Reply:
Ugh, that's right. I'll noodle on this for a bit